install_test.sh

```diff
@@ -1,17 +1,52 @@
-#!/bin/sh
+#!/bin/bash
 
-set -ev
+set -e
 
-# Lint.
-shellcheck -s sh ./*.sh
+PATH=$HOME/bin/:$PATH
+
+# Install shfmt.
+shfmt_version=v2.6.2
+case $(uname -s) in
+Darwin) shfmt_os="darwin" ;;
+*) shfmt_os="linux" ;;
+esac
+shfmt_url="https://github.com/mvdan/sh/releases/download/${shfmt_version}/shfmt_${shfmt_version}_${shfmt_os}_amd64"
+curl -sSL -o "$HOME/bin/shfmt" "$shfmt_url"
+chmod +x "$HOME/bin/shfmt"
+
+# Print versions.
+$SHELL --version
+curl --version
+shellcheck --version
+shfmt --version
+
+# Check formatting.
 shfmt -d .
 
-# Test we can install a specific version
-rm -rf ~/.deno
-./install.sh v0.2.0
-~/.deno/bin/deno -v | grep -e 0.2.0
+# Lint code.
+shellcheck -s sh ./*.sh
+shellcheck -s bash ./*.sh
+shellcheck -s dash ./*.sh
+shellcheck -s ksh ./*.sh
+
+test_specific_version() {
+	rm -rf ~/.deno
+	$1 install.sh v0.2.0
+	~/.deno/bin/deno -v | grep -e 0.2.0
+}
+
+test_latest_version() {
+	rm -rf ~/.deno
+	$1 install.sh
+	~/.deno/bin/deno -v
+}
+
+case $(uname -s) in
+Darwin) shells=(sh bash ksh zsh) ;;
+*) shells=(sh bash dash ksh zsh) ;;
+esac
 
-# Test we can get the latest version.
-rm -rf ~/.deno
-sh ./install.sh
-~/.deno/bin/deno -v
+for shell in $"${shells[@]}"; do
+	test_specific_version $shell
+	test_latest_version $shell
```

Comment:
This isn't necessary - we only need to test under sh.

Reply:
But `sh` on `osx` is just `bash`, right?

And `dash`, for example, doesn't support `set -o pipefail`.

Is this really not necessary?

Reply:
Perhaps it's not necessary, but it also doesn't hurt to test a bit more, right? :)

Reply:
Right now, again, CI is failing on `osx`, but working on `linux`. Yesterday, `linux` was failing for me, but `osx` was working.

I'd rather test more options than less. :)